mlflow/models/evaluation/default_evaluator.py

```diff
@@ -540,6 +573,17 @@ def plot_pr_curve():
 
         self._log_pandas_df_artifact(per_class_metrics_collection_df, "per_class_metrics")
 
+    def _evaluate_custom_metrics(self):
+        for custom_metric in self.custom_metrics:
+            # recreating the dataframe for each custom metric function call,
+            # in case the user modifies the eval_df inside their custom function.
```

Comment:
in theory, it's possible for `self.y_pred` / `self.y` to be an object that gets modified during `_evaluate_custom_metric`. Embedding the object in a new pandas dataframe won't help in that case. Perhaps we want to use `deepcopy()` on `self.y_pred` and `self.y`.

Reply:
Good catch!